packages/rollup-full-node/src/app/handler/routing-handler.ts

```diff
@@ -0,0 +1,171 @@
+/* External Imports */
+import { Address } from '@eth-optimism/rollup-core'
+import { getLogger, logError, SimpleClient } from '@eth-optimism/core-utils'
+
+import { parseTransaction, Transaction } from 'ethers/utils'
+/* Internal Imports */
+import {
+  FullnodeHandler,
+  InvalidParametersError,
+  InvalidTransactionDesinationError,
+  Web3RpcMethods,
+} from '../../types'
+import { AccountRateLimiter } from '../utils'
+
+const log = getLogger('routing-handler')
+
+const methodsToRouteWithTransactionHandler: string[] = [
+  Web3RpcMethods.sendRawTransaction,
+  Web3RpcMethods.sendTransaction,
+  Web3RpcMethods.getTransactionByHash,
+  Web3RpcMethods.getBlockByNumber,
+  Web3RpcMethods.getBlockByHash,
+]
+
+/**
+ * Handles rate-limiting requests by Ethereum address for transactions and by IP address for all
+ * other request types and then routes them according to their method.
+ *
+ * If they are read-only they'll go to the provided read-only provider, and
+ * otherwise they'll go to the transaction provider.
+ */
+export class RoutingHandler implements FullnodeHandler {
+  private readonly readOnlyClient: SimpleClient
+  private readonly transactionClient: SimpleClient
+
+  private readonly accountRateLimiter: AccountRateLimiter
+
+  constructor(
+    transactionHandlerUrl: string,
+    readonlyHandlerUrl: string,
+    maxRequestsPerTimeUnit: number,
+    maxTransactionsPerTimeUnit: number,
+    requestLimitPeriodInMillis: number,
+    private readonly deployAddress: Address,
+    private readonly toAddressWhitelist: Address[] = []
+  ) {
+    this.readOnlyClient = new SimpleClient(readonlyHandlerUrl)
+    this.transactionClient = new SimpleClient(transactionHandlerUrl)
+
+    if (
+      !!maxTransactionsPerTimeUnit &&
+      !!maxTransactionsPerTimeUnit &&
+      !!requestLimitPeriodInMillis
+    ) {
+      this.accountRateLimiter = new AccountRateLimiter(
+        maxRequestsPerTimeUnit,
+        maxTransactionsPerTimeUnit,
+        requestLimitPeriodInMillis
+      )
+    }
+  }
+
+  /**
+   * Handles the provided request by
+   * * Checking rate limits (and throwing if there's a violation)
+   * * Making sure that the destination address is allowed
+   * * Routing the request to the appropriate provider.
+   *
+   * @param method The Ethereum JSON RPC method.
+   * @param params The parameters.
+   * @param sourceIpAddress The requesting IP address.
+   */
+  public async handleRequest(
+    method: string,
+    params: any[],
+    sourceIpAddress: string
+  ): Promise<string> {
+    log.debug(
+      `Proxying request for method [${method}], params: [${JSON.stringify(
+        params
+      )}]`
+    )
+    let tx: Transaction
+
+    if (
+      method === Web3RpcMethods.sendTransaction ||
+      method === Web3RpcMethods.sendRawTransaction
+    ) {
+      try {
+        tx = parseTransaction(params[0])
```

Comment:
Having recently spent some time debugging/working on the `sendTransaction` endpoint, I can shed some color here--I believe this parsing will always fail for `eth_sendTransaction` requests.  The reason is because the standard JSON RPC parameters for that method are not actually equivalent to a transaction.

In general, I think the fullnode will only need expose the `sendTransaction` endpoint for the `test-handler` and never on a deployed instance.  CC @karlfloersch to sanity check that, but if so it should be sufficient to just remove `sendTransaction` endpoint.

Reply:
Ahh ok. That works. Is that merged in? Surprised tests pass, but I'll route it to the read-only one.

Reply:
Yep we should only expose `sendTransaction` on our test web3 handler & sendRawTransaction will be the only tx sending endpoint I believe for the demo

Reply:
Sounds good. Designated it as a test endpoint.